lib/SIL/Utils/OwnershipUtils.cpp

```diff
@@ -376,6 +397,13 @@ bool BorrowedValue::visitLocalScopeTransitiveEndingUses(
     function_ref<void(Operand *)> visitor) const {
   assert(isLocalScope());
 
+  if (kind == BorrowedValueKind::YieldedValue) {
+    // Just delegate to our visit local scope ending use visitor since yielded
+    // values are never consumed.
+    visitLocalScopeEndingUses(visitor);
```

Comment:
The changes here allow us to eliminate copy_value from dominated yielded guaranteed values. Before this patch our coroutine scopes were too narrow (i.e. we always immediately copied the value and then had the end_apply) so this would not have been useful. With the coroutine lifetime extension we see this case now. I need to add a test for this before this patch goes in (and TBH will most likely commit this separately).